In [1]:
from dotenv import load_dotenv
import os
import sys

sys.path.append('../')

load_dotenv()

config = {
    "base_url": os.getenv("BASE_URL"),
    "client_id": os.getenv("CLIENT_ID"),
    "client_secret": os.getenv("CLIENT_SECRET")
}


In [2]:
import re

def get_page(url):
    """
    Devuelve el número de página de una URL.
    """
    match = re.search(r'page=(\d+)', url)
    if match:
        return int(match.group(1))


In [4]:
from services.employment_records_mx_service import EmploymentRecordsMXService

employment_records_mx_service = EmploymentRecordsMXService(config)
all_employment_records = []
link_id = 'b73b86ce-c6cc-4b66-a8c8-f37b76cd7fa9'

page= 1

while page:
    employment_records = employment_records_mx_service.get_employment_records(link_id,page)
    all_employment_records.extend(employment_records['results'])
    page = get_page(employment_records['next']) if employment_records['next'] else None

In [5]:
len(all_employment_records)

1

In [7]:
record_id = all_employment_records[0]['id']

In [8]:
record_details = employment_records_mx_service.get_employment_records_details(record_id)

In [9]:
record_details

{'id': '5d0139c8-286f-4b1d-bbc3-5fa881885c4c',
 'link': 'b73b86ce-c6cc-4b66-a8c8-f37b76cd7fa9',
 'created_at': '2025-07-31T11:57:11.989755Z',
 'files': None,
 'report_date': '2025-07-31',
 'collected_at': '2025-07-31T11:57:11.249563Z',
 'personal_data': {'email': 'patricianaranjo@example.org',
  'last_name': 'Robles Quintana',
  'birth_date': '1991-02-18',
  'first_name': 'Joaquín',
  'document_ids': [{'document_type': 'NSS', 'document_number': '70041684898'},
   {'document_type': 'CURP', 'document_number': 'BLPM951331IONVGR54'}],
  'entitlements': {'status': 'UNKNOWN',
   'valid_until': None,
   'entitled_to_company_benefits': False,
   'entitled_to_health_insurance': False},
  'official_name': 'Joaquín Robles Quintana'},
 'employment_records': [{'state': 'DISTRITO FEDERAL',
   'currency': 'MXN',
   'employer': 'Laboratorios Sarabia-Lemus',
   'end_date': '2010-07-29',
   'start_date': '1997-09-29',
   'employer_id': '165-LAB-23981-CDMX',
   'collected_at': '2025-07-31T11:57:11.249157

# Mexican Employment Records EDA Summary

## Dataset Overview
- **Total Records**: 1 employment record
- **Report Date**: July 31, 2025
- **Data Source**: Mexican Social Security (IMSS) employment history

## Personal Information
- **Employee**: Joaquín Robles Quintana
- **Birth Date**: February 18, 1991
- **Email**: patricianaranjo@example.org
- **Documents**: 
  - NSS: 70041684898
  - CURP: BLPM951331IONVGR54

## Employment History

### Employment Details
- **Employer**: Laboratorios Sarabia-Lemus
- **Employer ID**: 165-LAB-23981-CDMX
- **Location**: Distrito Federal
- **Employment Period**: September 29, 1997 to July 29, 2010
- **Duration**: 669 weeks (~12.9 years)
- **Currency**: MXN (Mexican Peso)

### Salary Information
- **Monthly Salary**: $26,469.50 MXN
- **Most Recent Base Salary**: $870.23 MXN
- **Initial Base Salary**: $822.21 MXN (hired)
- **Final Base Salary**: $870.23 MXN (dismissed/resigned)

### Employment Status Updates
1. **HIRED** (Sep 29, 1997): Base salary $822.21 MXN
2. **SALARY_MODIFICATION** (Jun 17, 2005): Base salary $870.23 MXN
3. **SALARY_MODIFICATION** (Jan 12, 2009): Base salary $887.99 MXN
4. **DISMISSED_RESIGNED** (Jul 29, 2010): Base salary $870.23 MXN

## Social Security Summary
- **Weeks Contributed**: 669 weeks
- **Weeks Redeemed**: None
- **Weeks Reinstated**: None

## Benefits & Entitlements
- **Status**: Unknown
- **Health Insurance**: Not entitled
- **Company Benefits**: Not entitled
- **Valid Until**: Not specified

## Key Insights
- Long-term employment relationship spanning nearly 13 years
- Salary progression with multiple adjustments over time
- Complete employment lifecycle from hiring to departure
- Strong contribution record to social security system
- Employment ended in 2010, suggesting historical data analysis
- Single comprehensive record providing detailed employment timeline

## Data Quality
- Complete personal identification information
- Detailed salary history with specific dates
- Comprehensive employment status tracking
- Full social security contribution record
- No missing critical data points